src/components/SlateEditor/plugins/blockPicker/SlateBlockPicker.tsx

```diff
@@ -91,77 +182,64 @@ const SlateBlockPicker = ({
 
   const [selectedParagraph, selectedParagraphPath] = getCurrentBlock(editor, TYPE_PARAGRAPH) || [];
 
-  const selection = editor.selection;
-
-  const getLeftAdjust = () => {
-    const parent =
-      selectedParagraphPath && Editor.node(editor, Path.parent(selectedParagraphPath))?.[0];
-
-    if (Element.isElement(parent) && parent.type === TYPE_LIST_ITEM) {
-      return 110;
-    }
-    if (isTableCell(parent)) {
-      return 100;
-    }
-    if (Element.isElement(parent) && parent.type === TYPE_GRID) {
-      return -100;
-    }
-
-    return 78;
-  };
-
-  const show =
-    !isInTableCellHeader(editor, selectedParagraphPath) &&
-    isParagraph(selectedParagraph) &&
-    Node.string(selectedParagraph) === '' &&
-    selectedParagraph.children.length === 1 &&
-    selectedParagraphPath &&
-    selection &&
-    Path.isDescendant(selection.anchor.path, selectedParagraphPath) &&
-    Range.isCollapsed(selection);
-
-  const portalRef = createRef<HTMLDivElement>();
-
   useEffect(() => {
-    setTimeout(() => {
-      updateMenu();
-    }, 0);
-  });
+    const el = portalRef.current;
+    const { selection } = editor;
+    if (!el) return;
 
-  const updateMenu = () => {
-    const menu = portalRef.current;
-    if (!menu) {
-      return;
-    }
-    if (!show || !selectedParagraph) {
-      menu.removeAttribute('style');
-      return;
-    }
+    const [node] = Editor.nodes(editor, {
+      match: (node) => Element.isElement(node) && !editor.isInline(node),
+      mode: 'lowest',
+    });
 
-    if (!editor.shouldShowToolbar()) {
-      menu.removeAttribute('style');
+    const [illegalBlock] = Editor.nodes(editor, {
+      match: (node) => Element.isElement(node) && illegalAreas.includes(node.type),
+    });
+
+    if (
+      !selection ||
+      !selectedParagraph ||
+      isInTableCellHeader(editor, selectedParagraphPath) ||
+      Node.string(selectedParagraph) !== '' ||
+      !editor.shouldShowBlockPicker?.() ||
+      illegalBlock ||
+      (Element.isElement(node[0]) && !allowedPickAreas.includes(node[0].type))
+    ) {
+      el.style.display = 'none';
       return;
     }
-
+    const parent =
+      selectedParagraphPath && Editor.node(editor, Path.parent(selectedParagraphPath))?.[0];
+    const leftAdjust = getLeftAdjust(parent);
     const domElement = ReactEditor.toDOMNode(editor, selectedParagraph);
     const rect = domElement.getBoundingClientRect();
-
-    const left = rect.left + window.scrollX - getLeftAdjust();
-    menu.style.top = `${rect.top + window.scrollY - 14}px`;
-    menu.style.left = `${left}px`;
-  };
+    el.style.display = 'block';
+    const left = rect.left + window.scrollX - leftAdjust;
+    el.style.top = `${rect.top + window.scrollY - 14}px`;
+    el.style.left = `${left}px`;
+  });
```

Comment:
Er det en grunn til at ikke denne useEffecten har noen dep array?

typ noe sånt som `[allowedPickAreas, editor, illegalAreas, selectedParagraph, selectedParagraphPath]`


også: hva med å putte logikken i useEffecten her i en funksjon med et passende navn? hjelper nok litt på leseligheten!

Reply:
Ja og nei: Den kunne sikkert hatt et dep-array, men det er på denne måten den gamle block-pickeren er satt opp, og også sånn Slate gjør det i eksemplene sine. Hvis det er en ting som alltid burde rerendres vil jeg tørre å påstå at det er denne; Konsekvensen av å feilaktig vise block-pickeren er ganske stor. 

Ref funksjon: Øker det lesbarheten noe særlig? Det vil være like mye indentering. Dersom vi hadde hatt flere effekter ville jeg sagt meg enig, men denne komponenten har bare to.